cryostatio/cryostat-helm
# feat(podAnnotations): individualized Pod Annotations

Related to #220

To test:
1. ex. `helm install --set-json 'podAnnotations={"foo":"bar","com.example/annotation":"hello"}' --set-json 'storage.podAnnotations={"alice":"bob"}' cryostat ./charts/cryostat`
2. Wait for chart to deploy
3. Verify annotations are set:
```
$ oc get -o jsonpath='{.spec.template.metadata.annotations}' deployments/cryostat-v4-storage | jq
{
  "alice": "bob",
  "com.example/annotation": "hello",
  "foo": "bar"
}
```

```
$ oc get -o jsonpath='{.spec.template.metadata.annotations}' deployments/cryostat-v4 | jq
{
  "com.example/annotation": "hello",
  "foo": "bar"
}
```
4. Try to install with conflicting annotations: `helm install --set-json 'podAnnotations={"foo":"bar"}' --set-json 'core.podAnnotations={"foo":"baz"}' cryostat ./charts/cryostat`
5. Verify that the installation preferred the pod-specific annotation:
```
$ oc get -o jsonpath='{.spec.template.metadata.annotations}' deployments/cryostat-v4 | jq
{
  "foo": "baz"
}
```
